src/Domain/CoordinateMaps/TimeDependent/ShapeMapTransitionFunctions/Sphere.hpp

```diff
@@ -72,4 +72,16 @@ class SphereTransition final : public ShapeMapTransitionFunction {
   double b_{};
   static constexpr double eps_ = std::numeric_limits<double>::epsilon() * 100;
 };
+
+/// \cond
+// The Sphere class was renamed from SphereTransition at some point. This is
+// only an issue when it comes to pupping. Specifically deserializing, because
+// the name of the class is serialized along with its data. So if we only
+// register the Sphere derived class, but try to read in data that was
+// serialized with the SphereTransition name, then we will get an unregistered
+// PUP_ID error. Therefore, we add this class with the SphereTransition name
+// which is identical to the Sphere class for purposes of deserialization. This
+// class is not to be used otherwise.
+class SphereTransition : public Sphere {};
```

Comment:
Not super happy about this, but I would like to rename the class because the original name is redundant. If reviewers are strongly against this, I can forgo the name change

Reply:
The reason I added this was because of the versioning test in `Test_Domain` where we deserialize a BCO domain from H5 that was serialized with the old transition function name. Not sure if there's a good way to override charm registration name when reading something in that was already written. Open to suggestions

Reply:
I meant we could provide a way to override the second argument in
```c++
    PUPable_reg2(class_to_register,
                 registration_name<class_to_register>().c_str());
```
but if it's only for one thing probably not worth it.

Ping on the question about my_PUP_ID.

Reply:
I tried compiling with
```cpp
class SphereTransition : public Sphere {};
PUP::able::PUP_ID SphereTransition::my_PUP_ID = 0;
```
and got an error about `no member named 'my_PUP_ID'`. The goal is to have the `SphereTransition` class be identical to the `Sphere` class in every way except name, so I think it makes sense to not give it its own PUP_ID.

Reply:
Yes, you'd need to add the stuff in the class body as well.

> I think it makes sense to not give it its own PUP_ID.

But you are assigning two `PUP_ID`s, since you register twice, just one is overwriting the other.  Looking through the charm code I *think* this has the effect that you cannot serialize the first one registered through a base class pointer.  Or maybe you can but it is recorded as being the other type, not sure.  If you want to do this you definitely need tests that the serialization works the way you expect it to.

Reply:
Hmm ok. This is getting too complicated and I'd rather not run the risk of breaking serialization things like this, so I'll just forgo the name change. It's not a huge deal, I'll just name future transition functions properly. If there ever comes a time where we drop support for certain versions of maps, then we can revisit this.